src/domain/Customer.ts

```diff
@@ -0,0 +1,41 @@
+import { NotEligibleForMortgageException } from '../exceptions/NotEligibleForMortgageException';
+
+export class Customer {
+    id: string;
+    firstName: string;
+    lastName: string;
+    balance: number;
+    badCreditHistoryCount: number;
+
+    constructor(
+        id: string,
+        firstName: string,
+        lastName: string,
+        balance: number,
+        badCreditHistoryCount: number
+    ) {
+        this.id = id;
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.balance = balance;
+        this.badCreditHistoryCount = badCreditHistoryCount;
+    }
```

Comment:
We can simplify this part

```suggestion
    constructor(
        public id: number,
        public firstName: string,
        public lastName: string,
        public balance: number,
        public badCreditHistoryCount: number
    ) {}
```